src/api/c/median.cpp

```diff
@@ -36,7 +36,7 @@ static double median(const af_array& in) {
 
     af_array temp = 0;
     AF_CHECK(af_moddims(&temp, in, 1, dims.get()));
-    const Array<T> input = getArray<T>(temp);
+    const Array<T>& input = getArray<T>(temp);
```

Comment:
I think we can avoid such mistakes if we use auto in locations such as these where return type is obvious and evident from the function name.

Reply:
auto will also cause this because by default auto is by value which would result in the same behavior.

Reply:
yeah, my bad. forgot that even that needs `&` at the end.